test/gtest/common/test_helpers.h

```diff
@@ -476,6 +480,83 @@ class message_stream {
 } // ucs
 
 
+namespace ucp {
+
+class data_type_desc_t {
+public: 
+    data_type_desc_t()
+        : _origin(uintptr_t(NULL)), _length(0), _buf(NULL),
+          _iov_cnt_limit(sizeof(_iov) / sizeof(_iov[0])) {};
+
+    data_type_desc_t(ucp_datatype_t datatype, void *buf, size_t length)
+        : _origin(uintptr_t(buf)), _length(length), _buf(NULL),
+          _iov_cnt_limit(sizeof(_iov) / sizeof(_iov[0])) {
+        make(datatype, buf, length);
+    }
+
+    data_type_desc_t(ucp_datatype_t datatype, void *buf, size_t length,
+                     size_t iov_count)
+        : _origin(uintptr_t(buf)), _length(length), _buf(NULL),
+          _iov_cnt_limit(sizeof(_iov) / sizeof(_iov[0])) {
+        make(datatype, buf, length, iov_count);
+    };
+
+    data_type_desc_t &make(ucp_datatype_t datatype, void *buf, size_t length,
```

Comment:
i think "make" functions are supposed to be private?

Reply:
I'll add tests which do delayed initialization in next PR.

Reply:
i see..
why is make() returning *this? why not just void?

Reply:
just to reduce a bit code in usage like this
```
type &var = arr[i].make(...);
foo(var.a(), var.b(), var.c());
```
the same with `forward_to`